### PR TITLE
puts chem mass spectrometer inline with other machines

### PR DESCRIPTION
**Pull request by manofpepsi**
## About The Pull Request
Makes the chem_mass_spec a proper machine and puts it inline with other crew machinery by making it: buildable, deconstructable, movable and upgradable. In addition fixes a tiny grammar error in the description of the machine, adds a comment on the DO_NOT_SPLASH trait and removes some comments from the machine code and adds all of the code needed for the circuit board and the research node.

## Why It's Good For The Game

the chem mass spec is now an actual machine that is inline with how other machines work.

## Changelog
:cl: Pepsiman0
fix: the chem mass spectrometer is now a proper machine that is now: buildable, deconstructable, movable and upgradable. (the research for the board is in the basic bio tech node)
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/machinery/chem_mass_spec.dm [hunk 1]
```diff
@@ -32,36 +33,66 @@ This will not clean any inverted reagents. Inverted reagents will still be corre
 	var/obj/item/reagent_containers/beaker1
 	///Output reagents container
 	var/obj/item/reagent_containers/beaker2
+	///multiplies the final time needed to proccess the chems depending on the laser stock part
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/machinery/chem_mass_spec.dm:36]:
```suggestion
	///multiplies the final time needed to process the chems depending on the laser stock part
```

File: code/modules/reagents/chemistry/machinery/chem_mass_spec.dm [hunk 2]
```diff
@@ -32,36 +33,66 @@ This will not clean any inverted reagents. Inverted reagents will still be corre
...
+	var/CMS_coefficient = 1
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/machinery/chem_mass_spec.dm:37]: `cms_coefficient`

File: code/modules/reagents/chemistry/machinery/chem_mass_spec.dm [hunk 3]
```diff
@@ -32,36 +33,66 @@ This will not clean any inverted reagents. Inverted reagents will still be corre
...
 
 /obj/machinery/chem_mass_spec/Initialize()
 	. = ..()
-	beaker2 = new /obj/item/reagent_containers/glass/beaker/large(src)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/machinery/chem_mass_spec.dm:38]: This is an unwelcome behavioral change. Use the `mapload` variable on Initialize, and only add it then.

File: code/modules/reagents/chemistry/machinery/chem_mass_spec.dm [hunk 4]
```diff
@@ -32,36 +33,66 @@ This will not clean any inverted reagents. Inverted reagents will still be corre
...
+
+	if(default_deconstruction_screwdriver(user, icon_state, icon_state, item))
+		update_appearance()
+		return
+
+	if(default_deconstruction_crowbar(item))
+		return
+
+	if(default_unfasten_wrench(user, item))
+		return
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/machinery/chem_mass_spec.dm:82]: Put deconstruction under right-click (attackby_secondary).

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
I'll let this slide as a consistency fix--I remember ninjanomnom saying that every machine in the station should be constructable.

File: code/modules/reagents/chemistry/machinery/chem_mass_spec.dm (hunk 4 above)
Comment (manofpepsi) [reply code/modules/reagents/chemistry/machinery/chem_mass_spec.dm]: done

File: code/modules/reagents/chemistry/machinery/chem_mass_spec.dm (hunk 3 above)
Comment (manofpepsi) [reply code/modules/reagents/chemistry/machinery/chem_mass_spec.dm]: done

File: code/modules/reagents/chemistry/machinery/chem_mass_spec.dm (hunk 2 above)
Comment (manofpepsi) [reply code/modules/reagents/chemistry/machinery/chem_mass_spec.dm]: no more caps

**Comment by Mothblocks** [comment]
Unwrenching should probably be RMB too but this is good enough

**Pull request merged**